File: rust/array/217_contains_duplicate.rs


pub fn contains_duplicate(nums: Vec<i32>) -> bool {


    // hashmap or hashSet
    // use std::collections::HashMap;

    // let mut hmap = HashMap::new();
    // for &item in &nums{
    //     if let Some(_) = hmap.get(&item) {
    //         return false
    //     }
    //     hmap.insert(item, ());
    // }
    // false

    // hashSet
    use std::collections::HashSet;

    let mut hset = HashSet::new();

    for &item in nums.iter() {
        if hset.contains(&item) {
            return true
        }
        hset.insert(item);
    }
    false
}

File: go/array/217_contains_duplicate.go
package array

func containsDuplicate(nums []int) bool {
	// HashMap

	hmap := make(map[int]struct{}) // {val: struct{}}

	for _, v := range nums {
		if _, ok := hmap[v]; ok {
			return true
		}
		hmap[v] = struct{}{}
	}
	return false
}
